README.md

```diff
@@ -0,0 +1,701 @@
+    class Header {
+        version: u64,
+        schema_id: "revoke_v1"
+        signature: Signature,
+        public_key: PublicKey,
+        payload_hash: Option~Hash~,
+        payload_size: Option~u64~,
+        timestamp: u64,
+        seq_num: u64,
+    }
+    class `Body (Revocation)` {
+        revoke: Hash
+    }
+```
+
+# Validation
+
+Validation of capabilities must be performed by an authorizer before a any new operation arriving at a peer, or any sync request, is authorized.
+
+If any of the following criteria are not met, the capability must be considered invalid:
+
+## Time Bounds
+
+The system time must not be earlier than `not_before` or have passed the `expires` timestamp. Any new sync requests attempting to use this capability to read documents must be rejected and no operations relying on this capability for authority to perform an action must be rejected.
+
+In the case of any `write` capabilities expiring, a peer can choose to keep already accepted operations which rely on this capabilities authority after expiry has passed, or where possible, roll back these changes and remove the operations. Which approach is taken will depend on the desired behavior for the document.
+
+## Condition Attenuation
+
+Delegated conditions must not expand the scope of the capability it delegates from, it must only attenuate.
+
+The following table gives some examples of valid and invalid delegated conditions.
+
+| Received Conditions                                | Delegated Conditions                               | Valid | Reason                                        |
+| -------------------------------------------------- | -------------------------------------------------- | ----- | --------------------------------------------- |
+| `{document_ids: ["0X01", "0X02"]}`                 | `{document_ids: ["0X01"]}`                         | ✅    | Condition is attenuated                       |
+| `{schema_ids: ["events"]}`                         | `{schema_ids: ["events"], document_ids: ["0X01"]}` | ✅    | New condition added which restricts authority |
+| `{from_timestamp: 10, to_timestamp: 100}`          | `{from_timestamp: 50, to_timestamp: 80}`           | ✅    | Condition is attenuated                       |
+| `{schema_ids: ["events"], document_ids: ["0X01"]}` | `{schema_ids: ["events"]}`                         | ❌    | Condition removed                             |
+| `{document_ids: ["0X01"]}`                         | `{document_ids: ["0X01", "0X02"]}`                 | ❌    | Condition expanded                            |
+| `{from_timestamp: 50, to_timestamp: 80}`           | `{from_timestamp: 0, to_timestamp: 100}`           | ❌    | Condition expanded                            |
+
+## Principal Alignment
+
+In delegation, the `receiver` field of every proof must match the `issuer` field of the capability being delegated to. This alignment must form a chain back to the `subject` for each resource.
```

Comment:
What is a "proof"? Do you mean "capability"?

I understand I think you're describing a capability chain, but I read the first sentence multiple times and it didn't parse.

you're delegating to other capabilities?

Reply:
Yes, the proof is the previous capability in the delegation chain. I can re-write this to be clearer, thanks.